### ISSUE TITLE
Duktape doesn't implement typedarray.slice()

### ISSUE THREAD

**Issue by fatcerberus** [svaarala/duktape#995]
I don't know whether this is documented, but it surprised me at least, `TypedArray.prototype.slice()` doesn't appear to be supported.  The following code throws "undefined not callable" upon encountering the `.slice()` call:

```js
	var input = "Oh no, PIG! *MUNCH*";
	var enc = new TextEncoder();
	var dec = new TextDecoder('utf-8');
	var buffer = enc.encode(input);  // returns Uint8Array
	var output = "";
	for (var i = 0; i < buffer.length; ++i) {
		var slice = buffer.slice(i, i + 1);
		output += dec.decode(slice, { stream: true });
	}
	output += dec.decode();
```

**Comment by fatcerberus** [comment]
At first I thought maybe `.slice()` was from a later standard, but it's in ES6:
http://www.ecma-international.org/ecma-262/6.0/#sec-%typedarray%.prototype.slice

**Comment by svaarala** [comment]
That's not really a surprise - there are several typed array things which are not yet implemented. The base implementation is from Khronos typed array specification, with some semantic corrections from ES6. But as a whole the ES6 typed array is not completely implemented.

**Comment by svaarala** [comment]
Not *yet* completely implemented to be exact, the goal is of course to eventually implement the ES6 bindings.

**Comment by fatcerberus** [comment]
Mainly `.slice()` being missing surprised me because I remember seeing several comments in the code about buffer slices.  Maybe that was for something else (Node.js Buffer?)

**Comment by svaarala** [comment]
Node.js Buffer has a slice() method, and Khronos typed arrays have a .subarray() method.

**Comment by fatcerberus** [comment]
Ah, `.subarray()` is what I can use, thanks.  As you can see from the code above I was trying to torture test my `TextDecoder` implementation. :)

**Comment by svaarala** [comment]
A useful tracking issue would be "Add missing ES6 typed array features" and list all the missing bindings and semantics. I'll do that comparison tonight if I have enough time so that there's something documenting the current gap.

**Comment by fatcerberus** [comment]
Ah, I see the difference: `.subarray()` reuses the backing ArrayBuffer so that both refer to the same data, whereas `.slice()` makes a copy.  The former is actually more useful in this case.

**Comment by svaarala** [comment]
Yeah, the main difference is whether a copy is made or not. Btw, Duktape does implement `ArrayBuffer.prototype.slice()` which makes a copy.
